src/app/campaign/store/selectors/availability-allocation-actuals.selectors.ts

```diff
@@ -0,0 +1,101 @@
+import { createSelector } from '@ngrx/store';
+import { Availability, AvailabilityWeeklyRollup, AvailabilityRollup } from '../models';
+import { selectRoutedFlightAvailabilityEntities } from './availability.selectors';
+import { selectRoutedFlightAllocationPreviewEntities } from './allocation-preview.selectors';
+import { selectRoutedFlightId, selectRoutedLocalFlightZones } from './flight.selectors';
+
+export const selectAvailabilityRollup = createSelector(
+  selectRoutedFlightAvailabilityEntities,
+  selectRoutedFlightAllocationPreviewEntities,
+  selectRoutedFlightId,
+  selectRoutedLocalFlightZones,
+  (availabilityEntities, allocationPreviewEntities, flightId, zones): AvailabilityRollup[] => {
+    // filter availability entities by routed flight zones
+    const availabilityZones =
+      zones && zones.filter(zone => availabilityEntities[`${flightId}_${zone}`]).map(zone => availabilityEntities[`${flightId}_${zone}`]);
+
+    // group each zone by weeks
+    const zoneWeeks =
+      availabilityZones &&
+      availabilityZones.map((availability: Availability) => {
+        // Each week is keyed by a date string of the week begin date. The days are accumulated onto it.
+        let weekBeginString: string;
+        let weekEnd: Date;
+        // reduce to acc weeks
+        return availability.days.reduce(
```

Comment:
Not sure how much of this is new vs moved ... but I appreciate the good commenting/tests on this mega function! :+12:

Reply:
yay comment! The model is updated to be more clear but otherwise not really too changed on these rollups. I did notice where I wasn't using UTC to find the end of week boundary, but that should be the only functional change.